core/src/main/java/org/elasticsearch/action/index/IndexRequest.java

```diff
@@ -606,7 +615,11 @@ public void writeTo(StreamOutput out) throws IOException {
     public String toString() {
         String sSource = "_na_";
         try {
-            sSource = XContentHelper.convertToJson(source, false);
+            if (source.length() > MAX_SOURCE_LENGTH_IN_TOSTRING) {
+                sSource = "too big: " + ByteSizeValue.formatBytesSizeValue(source.length());
```

Comment:
For perspective, the complete `toString()` output will look like this:
`index {[index][type][null], source[n/a, actual length: [12.7kb], max length: 2kb]}`